ledger-core/virtual/integration/outgoing_test.go

```diff
@@ -105,52 +158,69 @@ func TestVirtual_CallMethodOutgoing_WithPulseChange(t *testing.T) {
 		runnerMock.AddExecutionClassify("Foo", fooIsolation, nil)
 	}
 
-	callCount := 0
-
-	firstPulse := server.GetPulse().PulseNumber
-
-	expectedToken := payload.CallDelegationToken{
-		TokenTypeAndFlags: payload.DelegationTokenTypeCall,
-		PulseNumber:       server.GetPulse().PulseNumber,
-		Callee:            objectAGlobal,
-		Outgoing:          outgoingCallRef,
-		DelegateTo:        server.JetCoordinatorMock.Me(),
-	}
-
+	// add checks to typedChecker
 	{
 		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
-			assert.Equal(t, firstPulse, report.OrderedPendingEarliestPulse)
-			assert.Equal(t, int32(1), report.OrderedPendingCount)
+			// check for pending counts must be in tests: call constructor/call terminal method
+			assert.Equal(t, objectAGlobal, report.Object)
+			assert.Zero(t, report.DelegationSpec)
 			return false
 		})
 		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
-			assert.Zero(t, request.DelegationSpec)
+			assert.Equal(t, objectAGlobal, request.Callee)
+
+			msg := payload.VDelegatedCallResponse{
+				Callee: request.Callee,
+				ResponseDelegationSpec: payload.CallDelegationToken{
+					TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+					PulseNumber:       server.GetPulse().PulseNumber,
+					Callee:            request.Callee,
+					Outgoing:          request.CallOutgoing,
+					DelegateTo:        server.JetCoordinatorMock.Me(),
+				},
+			}
 
-			server.SendPayload(ctx, &payload.VDelegatedCallResponse{
-				Callee:                 request.Callee,
-				ResponseDelegationSpec: expectedToken,
-			})
+			if vdelegatedcallrequestCount == 0 {
```

Comment:
I'm not sure that is good practice:
1. you can rename that to "firstVDelegatedCallRequest" and use it bool if we should separate first and other message 
2. for only two times you can increment `VDelegatedCallRequestCount` every time and make switch for three cases 0, 1 and default (will lead to `t.Fatal`)

Also, pls, use SnakeCase for names of variables

Reply:
done